Services/Imprint/classes/class.ilObjLegalNoticeGUI.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\UI\Component\Component;
+
+/**
+ * @author            Michael Jansen <mjansen@databay.de>
+ * @ilCtrl_Calls      ilObjLegalNoticeGUI: ilPermissionGUI, ilImprintGUI
+ * @ilCtrl_isCalledBy ilObjLegalNoticeGUI: ilAdministrationGUI
+ */
+class ilObjLegalNoticeGUI extends ilObject2GUI
+{
+    private ilImprintGUI $legal_notice_gui;
+
+    public function __construct(int $a_id = 0, int $a_id_type = self::REPOSITORY_NODE_ID, int $a_parent_node_id = 0)
+    {
+        global $DIC;
+        $this->legal_notice_gui = new ilImprintGUI();
+        parent::__construct($a_id, $a_id_type, $a_parent_node_id);
+        $this->legal_notice_gui->setPresentationTitle($this->lng->txt("adm_imprint"));
+    }
+
+    public function getType(): string
+    {
+        return 'impr';
+    }
+
+    public function executeCommand(): void
+    {
+        $this->prepareOutput();
+
+        $nextClass = $this->ctrl->getNextClass($this);
+        $cmd = $this->ctrl->getCmd();
+
+
+        switch (strtolower($nextClass)) {
+            case strtolower(ilPermissionGUI::class):
+                $perm_gui = new ilPermissionGUI($this);
+                $this->ctrl->forwardCommand($perm_gui);
+                break;
+            default:
+                if (in_array($cmd, ['', 'view', 'edit', null], true) || !method_exists($this, $cmd)) {
```

Comment:
Why do we need `|| !method_exists($this, $cmd)`?

Reply:
https://github.com/ILIAS-eLearning/ILIAS/pull/5791/commits/e19bb712dceb890a49a15dbd2c47c7e9d97c2166#diff-3dd562884774a8008bdf2403959ed9dcfdabed7a50eebf62b5823333c8ff4abaR58-R69
this seems better